src/instance.rs

```diff
@@ -96,6 +96,221 @@ where
     }
 }
 
+impl<'py, T> Bound<'py, T> {
+    /// Downcast this `PyAny` to a concrete Python type or pyclass.
```

Comment:
The wording seems incorrect now that the receiver is `Bound<T>` instead of `&PyAny` or even `Bound<PyAny>`?

Reply:
Maybe "Downcast this Python value to a subtype"?

Reply:
As a second thought, it's not just about "downcasting" any more, as `U` might be a supertype of `T`. Would it be better to call these APIs `.cast()` instead of `.downcast()`?

(That would be a solution to avoid needing to remove the `PyAnyMethods`.)

I quite like `.cast()` though that might be a good reason to also leave this API until 0.22 and make it a more deliberate & documented addition.